server/schedule/event_handler.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package schedule
+
+import (
+	"context"
+	"time"
+
+	"github.com/CeresDB/ceresdbproto/pkg/metaservicepb"
+	"github.com/pkg/errors"
+)
+
+type Event interface {
+	ToHeartbeatResp() *metaservicepb.NodeHeartbeatResponse
+}
+
+type NoneEvent struct{}
+
+func (*NoneEvent) ToHeartbeatResp() *metaservicepb.NodeHeartbeatResponse {
+	// TODO: add response header
+	return &metaservicepb.NodeHeartbeatResponse{Timestamp: uint64(time.Now().UnixMilli())}
+}
+
+type OpenEvent struct {
+	ShardIDs []uint32
+}
+
+func (open *OpenEvent) ToHeartbeatResp() *metaservicepb.NodeHeartbeatResponse {
+	// TODO: add response header
+	return &metaservicepb.NodeHeartbeatResponse{Timestamp: uint64(time.Now().UnixMilli()), Cmd: &metaservicepb.NodeHeartbeatResponse_OpenCmd{
+		OpenCmd: &metaservicepb.OpenCmd{
+			ShardIds: open.ShardIDs,
+		},
+	}}
+}
+
+type EventHandler struct {
```

Comment:
`EventHandler` seems defined for `coordinator` and shall we put to `cluster` package?

Reply:
HeartBeatStreams need metaservicepb.NodeHeartbeatResponse as input, EventHandler abstract `Event` for cluster dynamic sheduling in ceresmeta.